### ISSUE TITLE
Auto-correct for common tpyos...

### ISSUE THREAD

**Issue by Stevefb** [rubberduck-vba/Rubberduck#4173]
Loving the new auto correct feature - hats off.  

Sometimes I accidentally type:  `If true them` when I really mean: `If true then`.  Anyway, I just _thought_ I had one of those moments and _thought_ I saw Rubberduck auto-magically change *them* to *Then* when I pressed enter.  

But Rubberduck didn't change anything so apparently I didn't make a typo (screen flicker?) but, all the same I thought it would be a nice addition :-)  

Not sure how realistic this is, but thought it might be worth mentioning. Some of my common typos:

- Them _instead of_ Then
- End It _instead of_ End If
- Dowhile _instead of_ Do While
- selecion _instead of_ Selection

Or, perhaps a custom typo dictionary could be added... hmmm...

**Comment by daFreeMan** [comment]
I have symptoms of selective lysdexia and often type `VyBal` instead of `ByVal`...

I believe that there are discussions of code snippets/short-cuts that could be repurposed for this if one were allowed to define ones own snippets.

**Comment by Stevefb** [comment]
@daFreeMan Agreed. I've been using MZ Tools Code Library with expansion shortcut it works well - when I remember to use it. Was also trying to learn C# so played in VS for a while and thought I'd gone bat s..t crazy when I returned to VBA to discover auto-correct... Hopefully RD could bring something similar but without having to press the expansion shortcut... (_just pressing SPACE &/or ENTER would be awesome_).

**Comment by retailcoder** [comment]
The AC behavior I'm thinking of for this will be the same as for block completion: the handler will select the entire "trigger word", and capture the next keypress - if it's {TAB} or {ENTER} (as per preferences/settings), autocorrect takes over; if it's any other key, selection is repositioned immediately after the "trigger word", and the keypress goes through.
